enhancements/installer/operators-check.md

```diff
@@ -0,0 +1,174 @@
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - @wking
+  - @LalatenduMohanty
+approvers:
+  - @sdodson
+  - @deads2k
+  - @bparees
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2021-07-14
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - none
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+  - n/a
+superseded-by:
+  - n/a
+---
+
+# Cluster Operator Installation Check
+
+## Summary
+
+This enhancement proposes that the OpenShift Installer should check the status
+of individual cluster operators and use those statuses as a threshold for 
+determining whether an install is a success (exit 0) or failure (exit != 0).
+Specifically, the Installer should check whether cluster operators have stopped
+progressing for a certain amount of time. If the Installer sees that an operator
+is Available=true but fails to enter a stable Progressing=false state, the Installer
+will exit non-zero and log an error.
+
+## Motivation
+
+This enhancement allows the Installer to identify and catch a class of errors
+which are currently going unchecked. Without this enhancement, the Installer can
+(and does) declare clusters successfully installed when the cluster has failing components.
+Adding a check for cluster operator stability will allow normal users and managed services teams
+to more easily identify and troubleshoot issues when they encounter this class of failure. The change
+will also help the Technical Release Team identify this class of problem in CI and triage
+issues to the appropriate operator development team.
+
+### User Stories
+
+As an openshift-install user, I want the Installer to tell me when an operator never stops
+progressing so that I can check whether the operator has an issue.
+
+As a member of the technical release team, I want the Installer to exit non-zero when
+an operator never stops progressing so that I can triage operator bugs.
+
+### Goals
+
+* Installer correctly identifies a failed cluster install where cluster operators are not stable
+* Installer exits non-zero in those situations and delivers meaningful error message
+
+### Non-Goals
+
+* Installer handling other classes of errors, such as failure to provision machines by MAO
+
+## Proposal
+
+### Workflow Description
+
+Cluster admin will begin installing cluster as usual. The Installation workflow will be:
+
+1. Cluster initializes as usual
+2. As usual, installer checks that cluster version is Available=true Progressing=False Degraded=False 
+3. Installer checks status of each cluster operator for stability
```

Comment:
For reference, here's the things we are currently looking at today with OSD/ROSA to try and determine when a cluster install has actually finished:
https://github.com/openshift/osde2e/blob/71f4013df9420fc961104a9388e42bc7c6af9b2f/pkg/common/cluster/clusterutil.go#L430-L470